executing/_position_node_finder.py

```diff
@@ -1,28 +1,30 @@
 import ast
 import dis
-from .executing import NotOneValueFound, only, function_node_types, assert_
+from enum import EnumMeta
+from types import CodeType, FrameType
+from typing import Any, Callable, Iterator, Optional, Sequence, Set, Tuple, Type, Union, cast
+from .executing import EnhancedAST, NotOneValueFound, Source, only, function_node_types, assert_
 from ._exceptions import KnownIssue, VerifierFailure
 
 from functools import lru_cache
 
 # the code in this module can use all python>=3.11 features
 
 
-def parents(node):
+def parents(node: EnhancedAST) -> Iterator[EnhancedAST]:
     while True:
         if hasattr(node, "parent"):
-            node = node.parent
+            node = cast(EnhancedAST, node.parent)
```

Comment:
This cast shouldn't be needed now that EnhancedAST.parent is also EnhancedAST. There's a few other similar casts.

Reply:
Fixed, and added the mypy warning for unnecessary casts (specifically, the case where you cast to something a variable already is)